js/src/evaluation/llm_evaluator.ts

```diff
@@ -0,0 +1,248 @@
+import { initChatModel } from "langchain/chat_models/universal";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import * as uuid from "uuid";
+import { EvaluationResult, EvaluationResults, RunEvaluator } from "./evaluator.js";
+import type { Run, Example } from "../schemas.js";
+export class CategoricalScoreConfig {
+  key: string;
+  choices: string[];
+  description: string;
+  includeExplanation: boolean;
+  explanationDescription?: string;
+
+  constructor(params: {
+    key: string;
+    choices: string[];
+    description: string;
+    includeExplanation?: boolean;
+    explanationDescription?: string;
+  }) {
+    this.key = params.key;
+    this.choices = params.choices;
+    this.description = params.description;
+    this.includeExplanation = params.includeExplanation ?? false;
+    this.explanationDescription = params.explanationDescription;
+  }
+}
+  
+export class ContinuousScoreConfig {
+  key: string;
+  min: number;
+  max: number;
+  description: string;
+  includeExplanation: boolean;
+  explanationDescription?: string;
+
+  constructor(params: {
+    key: string;
+    description: string;
+    min?: number;
+    max?: number;
+    includeExplanation?: boolean;
+    explanationDescription?: string;
+  }) {
+    this.key = params.key;
+    this.min = params.min ?? 0;
+    this.max = params.max ?? 1;
+    this.description = params.description;
+    this.includeExplanation = params.includeExplanation ?? false;
+    this.explanationDescription = params.explanationDescription;
+  }
+}
+
+type ScoreConfig = CategoricalScoreConfig | ContinuousScoreConfig;
+
+function createScoreJsonSchema(
+  scoreConfig: ScoreConfig,
+  reasoningKey: string
```

Comment:
Ditto above, but seems like a better API to make opting in to CoT only take one argument rather than 2 (split across two things)